addons/docs/src/frameworks/vue/sourceDecorator.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_vnode"] }] */
+
+import { addons, StoryContext } from '@storybook/addons';
+import { logger } from '@storybook/client-logger';
+import prettier from 'prettier/standalone';
+import prettierHtml from 'prettier/parser-html';
+import Vue from 'vue';
+
+import { SourceType, SNIPPET_RENDERED } from '../../shared';
+
+export const skipJsxRender = (context: StoryContext) => {
+  const sourceParams = context?.parameters.docs?.source;
+  const isArgsStory = context?.parameters.__isArgsStory;
+
+  // always render if the user forces it
+  if (sourceParams?.type === SourceType.DYNAMIC) {
+    return false;
+  }
+
+  // never render if the user is forcing the block to render code, or
+  // if the user provides code, or if it's not an args story.
+  return !isArgsStory || sourceParams?.code || sourceParams?.type === SourceType.CODE;
+};
+
+export const sourceDecorator = (storyFn: any, context: StoryContext) => {
+  const story = storyFn();
+
+  // See ../react/jsxDecorator.tsx
+  if (skipJsxRender(context)) {
+    return story;
+  }
+
+  try {
+    // Creating a Vue instance each time is very costly. But we need to do it
+    // in order to access VNode, otherwise vm.$vnode will be undefined.
+    // Also, I couldn't see any notable difference from the implementation with
+    // per-story-cache.
+    // But if there is a more performant way, we should replace it with that ASAP.
+    const vm = new Vue({
+      data() {
+        return {
+          STORYBOOK_VALUES: context.args,
+        };
+      },
+      render(h) {
+        return h(story);
+      },
+    }).$mount();
+
+    const channel = addons.getChannel();
+
+    const storyComponent = getStoryComponent(story.options.STORYBOOK_WRAPS);
+
+    const storyNode = lookupStoryInstance(vm, storyComponent);
+
+    const code = vnodeToString(storyNode._vnode);
+
+    channel.emit(
+      SNIPPET_RENDERED,
+      (context || {}).id,
+      prettier.format(`<template>${code}</template>`, {
+        parser: 'vue',
+        plugins: [prettierHtml],
+        // Because the parsed vnode missing spaces right before/after the surround tag,
+        // we always get weird wrapped code without this option.
+        htmlWhitespaceSensitivity: 'ignore',
```

Comment:
I'm using Prettier here just because "I know how to format code with Prettier". If another lib/approach is preferable, please tell me.